### ISSUE TITLE
Better error system

### ISSUE THREAD

**Issue by petrpan26** [hicder/muopdb#21]
Right now the codebase are full of unwrap of option and error. This will cause Rust to panic which is an undesirable behaviour. In Rust, panics are generally reserved for unrecoverable error. A suggestion is to have a result framework in place to better catch error.

We can either build a few simple error system then use std::Error or utilizing library such as https://docs.rs/anyhow/latest/anyhow/

**Comment by hicder** [comment]
yeah we definitely don't want those `unwrap`. let's move them to `Result`
